### PR TITLE
Implement v0.3.0 changes to the Standard

### PR DESCRIPTION
**Pull request by bfabio**
Implement v0.3.0 [changes to the Standard](https://github.com/publiccodeyml/publiccode.yml/pulls?q=is%3Apr+is%3Aclosed+label%3Avote-approved)

The only missing change will be addressed [in this PR](https://github.com/italia/publiccode-parser-go/pull/115)

Since the API has changed to implement warnings, this will warrant a new major release.

### CODE REVIEW

File: keys.go [hunk 1]
```diff
@@ -8,14 +8,14 @@ import (
...
+	urlutil "github.com/italia/publiccode-parser-go/v3/internal"
 )
 
 // validateFields validates publiccode with additional rules not validatable
 // with a simple YAML schema.
-// It returns any error encountered as ValidationErrors.
+// It returns any error encountered as ValidationResults.
 func (p *Parser) validateFields() error {
-	var ve ValidationErrors
+	var ve ValidationResults
```
Comment (sebbalex) [inline keys.go:18]: I'd rename this variable as `vr` to be consistent

**Review by sebbalex — APPROVED** [review]

File: pcvalidate/pcvalidate.go [hunk 2]
```diff
@@ -94,8 +94,12 @@ func main() {
 	} else {
 		if err != nil {
 			fmt.Println(err)
+		}
+		if hasValidationErrors(err) {
 			os.Exit(1)
```
Comment (LorenzoS92) [inline pcvalidate/pcvalidate.go:99]: Can we return the error instead of exiting brutally here?

**Review by LorenzoS92 — APPROVED** [review]

File: pcvalidate/pcvalidate.go (hunk 2 above)
Comment (bfabio) [reply pcvalidate/pcvalidate.go:99]: This is a command line tool, once we get there we already printed everything we needed and we have to exit (with status)

Am I missing something?

File: parser.go [hunk 3]
```diff
@@ -280,10 +303,15 @@ func (p *Parser) ParseBytes(in []byte) error {
 
 	err = p.validateFields()
 	if err != nil {
-		for _, err := range err.(ValidationErrors) {
-			err.Line, err.Column = getPositionInFile(err.Key, node)
-
-			ve = append(ve, err)
+		for _, err := range err.(ValidationResults) {
```
Comment (bfabio) [inline parser.go:306]: I'm not really sure if using a type for a slice of `error`s is the kosher thing to do in Go, but the parser did it before with `ValidationErrors` :shrug:

Comment (LorenzoS92) [reply parser.go:306]: No it's, ok, i've already seen that when dealing with external's APIs error responses that come from an SDK.

**Pull request merged**